Documentation updates

Slayer95 opened the pull request.

Zarel commented on config/config-example.js:
```diff
@@ -73,10 +75,17 @@ exports.reportjoins = true;
-// Turn this off on large tournament servers where battles get a lot of joins and leaves.
+// report joins and leaves in battle - shows messages like "<USERNAME> joined"
+//   This feature can clutter battle logs - turn this off if your server hosts
+//   battles with more than 30 or so spectators (such as big tournament battles).
```
This should really not be done this way; you should detect how many users are in the battle and use that to determine whether or not to report battlejoins.

Slayer95 replied:
So, should I remove this from the docs ?_?

Zarel replied:
I guess keep it, that _is_ how it works right now.

Zarel replied:
Except is this setting ever off? It shouldn't be...

Zarel replied:
Yeah, actually we should remove this feature entirely. The fact that some large battles exist should have zero negative effect on the user experience for small battles.

Slayer95 replied:
Wait, what... so should it be de facto always on or off? :|

(Remember that this was requested by chaos for Smogtours...)

Zarel replied:
Oh, I was going to do it properly later.

I'd say keep this undocumented and out of `config.js` until I fix it to not suck.

Zarel commented on config/config-example.js:
```diff
@@ -73,10 +75,17 @@ exports.reportjoins = true;
+//   only be able to see the messages if they have the /showjoins client-side setting enabled.
+//   Set this to a positive amount of milliseconds if you want to enable this feature.
+exports.reportjoinsperiod = 0;
+
```
They're not supposed to be seen at all.

The batches are to save bandwidth, it's not supposed to have any effect on how they're seen since reportjoins is supposed to be off.

Slayer95 replied:
"They're not supposed to be seen at all".
Tell that to people who like to have /showjoins all the time :|

Zarel commented on config/config-example.js:
```diff
@@ -84,6 +93,11 @@ exports.chatmodchat = false;
 exports.battlemodchat = false;
 exports.pmmodchat = false;
 
+// forced timer - sets the timer on for all battles
```
I'd say "force the timer on for all battles".

Slayer95 replied:
Okay

Slayer95 commented:
Updated

Zarel commented on config/config-example.js:
```diff
@@ -64,19 +64,23 @@ exports.loginserverpublickey = "-----BEGIN RSA PUBLIC KEY-----\n" +
 
-// report joins and leaves in battle - shows messages like "<USERNAME> joined" in battle
-// Turn this off on large tournament servers where battles get a lot of joins and leaves.
-exports.reportbattlejoins = true;
```
Wait, this is structured so not having it makes it default to `false`? Blah, we really need to fix this some other way.

Zarel replied:
I guess for now, we add that back. I'll remove it when I make that feature not suck.

Joimer replied:
Can be changed to ignorebattlejoins so battle joins are reported on by default, then you need to update at the same time smogtours, though.

Zarel replied:
So can you add this back, @Slayer95? I don't want to have to manual merge this. :|

Slayer95 commented:
Updated again. Also fixed the credit issue described in https://github.com/Zarel/Pokemon-Showdown/commit/028162dc5f0ede7aa7358b9a3c4605b8a3f3497e as it's pertinent to the PR.

Zarel commented on protocol-doc.md:
```diff
@@ -13,9 +13,9 @@ Showdown directly using WebSocket:
 Client implementations you might want to look at for reference include:
 
-- pickdenis's chat bot (Ruby) -
+- pickdenis' chat bot (Ruby) -
```
PS uses my own grammatical rules, which uses `s's` for possessive names.

Zarel replied:
(Which is a valid way of doing it, incidentally)
